### PR TITLE
Cover/smbiosPart2

### PR DESCRIPTION
**Pull request by walterchris**

### CODE REVIEW

File: pkg/smbios/struct_parser_test.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+			}
+
+			off, err := parseStruct(&table, 0, tt.complete, TPMDev)
+			if err != tt.wantErr {
+				if !strings.Contains(err.Error(), tt.wantErr.Error()) {
+					t.Errorf("Test %q: parseStruct() = %d, '%v' want '%v'", tt.name, off, err, tt.wantErr)
+				}
+			}
+
+			// TODO: Manual checking if it has been parsed correctly.
```
Comment (ChriMarMe) [inline pkg/smbios/struct_parser_test.go:170]: Was that ToDo a reminder?

**Review by ChriMarMe — APPROVED** [review]
LGTM. Schau dir nur den einen Kommentar an :)

File: pkg/smbios/struct_parser_test.go (hunk 1 above)
Comment (walterchris) [reply pkg/smbios/struct_parser_test.go:170]: Oh yeah - need to resolve this :) The questions is, is the check sufficient or do we need to check all fields manually?

File: pkg/smbios/info_linux_test.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+func TestFromSysfs(t *testing.T) {
+	testutil.SkipIfNotRoot(t)
+
+	if runtime.GOARCH != "amd64" {
+		t.Skip("Test only supported in amd64 Qemu")
+	}
+
+	info, err := FromSysfs()
+	if err != nil || info == nil {
+		t.Errorf("FromSysfs() = '%q', '%v', want nil", info, err)
```
Comment (llogen) [inline pkg/smbios/info_linux_test.go:22]: '%q' double quoted? I think %q should be enough

File: pkg/smbios/info_test.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+	}
+
+	info, err := FromSysfs()
+	if err != nil {
+		t.Errorf("FromSysfs as a requirement failed.")
+	}
+
+	systeminfo, err := info.GetSystemInfo()
+	if err != nil || systeminfo == nil {
+		t.Errorf("GetSystemInfo() = '%q', '%v'", systeminfo, err)
```
Comment (llogen) [inline pkg/smbios/info_test.go:45]: same

File: pkg/smbios/struct_parser_test.go [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+	UnknownType := &UnknownTypes{
+		Table: table,
+	}
+
+	off, err := parseStruct(&table, 0, false, UnknownType)
+	if err == nil {
+		t.Errorf("TestParseStructUnsupported : parseStruct() = %d, '%v' want: '%s'", off, err, want)
+	} else {
+		if !strings.Contains(err.Error(), want) {
+			t.Errorf("TestParseStructUnsupported : parseStruct() = %d, '%v' want: '%s'", off, err, want)
```
Comment (llogen) [inline pkg/smbios/struct_parser_test.go:44]: same

File: pkg/smbios/type17_memory_device_test.go [hunk 5]
```diff
@@ -0,0 +1,439 @@
...
+			MemDevFormFactor: 0x80,
+			want:             "0x80",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := tt.MemDevFormFactor.String()
+			if got != tt.want {
+				t.Errorf("MemDevFormFactor.String() = '%s', want '%s'", got, tt.want)
```
Comment (llogen) [inline pkg/smbios/type17_memory_device_test.go:86]: %q

File: pkg/smbios/type17_memory_device_test.go [hunk 6]
```diff
@@ -0,0 +1,439 @@
...
+			MemDevTechnology: 0x80,
+			want:             "0x80",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := tt.MemDevTechnology.String()
+			if got != tt.want {
+				t.Errorf("MemDevTechnology.String() = '%s', want '%s'", got, tt.want)
```
Comment (llogen) [inline pkg/smbios/type17_memory_device_test.go:152]: %q

File: pkg/smbios/type17_memory_device_test.go [hunk 7]
```diff
@@ -0,0 +1,439 @@
...
+			MemDevTypeDetail: MemoryDeviceTypeDetailWindowDRAM,
+			want:             "Window DRAM",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := tt.MemDevTypeDetail.String()
+			if got != tt.want {
+				t.Errorf("MemDevTypeDetail.String() = '%s', want '%s'", got, tt.want)
```
Comment (llogen) [inline pkg/smbios/type17_memory_device_test.go:190]: %q

File: pkg/smbios/type17_memory_device_test.go [hunk 8]
```diff
@@ -0,0 +1,439 @@
...
+			MemDevOpsCaps: MemoryDeviceOperatingModeCapabilityUnknown,
+			want:          "Unknown",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := tt.MemDevOpsCaps.String()
+			if got != tt.want {
+				t.Errorf("MemoryDeviceOperatingModeCapability.String() = '%s', want '%s'", got, tt.want)
```
Comment (llogen) [inline pkg/smbios/type17_memory_device_test.go:228]: %q

File: pkg/smbios/type17_memory_device_test.go [hunk 9]
```diff
@@ -0,0 +1,439 @@
...
+	Cache Size: 4 kB
+	Logical Size: 4 kB`,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := tt.MemDev.String()
+			if got != tt.want {
+				t.Errorf("MemoryDevice.String() = '%s', want '%s'", got, tt.want)
```
Comment (llogen) [inline pkg/smbios/type17_memory_device_test.go:435]: %q

**Review by llogen — CHANGES_REQUESTED** [review]
I just found nits. Lgtm!

**Review by llogen — APPROVED** [review]

**Pull request closed without merging**